bin/zowe-configure-component.sh

```diff
@@ -0,0 +1,159 @@
+    elif [[ -f "$COMPONENT_PATH/manifest.json" ]]; then
+        MANIFEST_PATH="$COMPONENT_PATH/manifest.json"
+    fi
+
+    COMMANDS_CFG_INSTANCE=$(eval "java -jar ${ZOWE_ROOT_DIR}/bin/utils/format-converter-cli.jar $MANIFEST_PATH | java -jar ${ZOWE_ROOT_DIR}/bin/utils/jackson-jq-cli.jar -r '.commands.configureInstance'")
+
+    if [ ! "$COMMANDS_CFG_INSTANCE" = "null" ]; then
+        ./$COMMANDS_CFG_INSTANCE
+    fi
+
+    DESKTOP_PLUGIN=$(eval "java -jar ${ZOWE_ROOT_DIR}/bin/utils/format-converter-cli.jar $MANIFEST_PATH | java -jar ${ZOWE_ROOT_DIR}/bin/utils/jackson-jq-cli.jar -r '.desktopPlugin'")
+
+    if [ ! "$DESKTOP_PLUGIN" = "null" ]; then
+            DESKTOP_PLUGIN_PATH=$(eval "java -jar ${ZOWE_ROOT_DIR}/bin/utils/format-converter-cli.jar $MANIFEST_PATH | java -jar ${ZOWE_ROOT_DIR}/bin/utils/jackson-jq-cli.jar -r '.desktopPlugin[].path'")
+            if [ ! "$DESKTOP_PLUGIN_PATH" = "null" ]; then
+                install_desktop_plugin
+            fi
+    fi
+
+    COMMANDS_START=$(eval "java -jar ${ZOWE_ROOT_DIR}/bin/utils/format-converter-cli.jar $MANIFEST_PATH | java -jar ${ZOWE_ROOT_DIR}/bin/utils/jackson-jq-cli.jar -r '.commands.start'")
+
+    if [ ! "$COMMANDS_START" = "null" ] && [ $IS_NATIVE = false ]; then
+        install_component
+    fi
+
+}
+
+while [ $# -gt 0 ]; do #Checks for parameters
+    arg="$1"
+    case $arg in
+        -o|--component) #Represents the path pointed to the component's compressed file
+            shift
+            COMPONENT_NAME=$1
+            shift
+        ;;
+        -i|--instance_dir) #Represents the path to zowe's instance directory (optional)
+            shift
+            path=$(get_full_path "$1")
+            validate_directory_is_accessible "$path"
+            if [[ $? -eq 0 ]]; then
+                validate_file_not_in_directory "$path/instance.env" "$path"
+                if [[ $? -ne 0 ]]; then
+                    INSTANCE_DIR=$path
+                else
+                    error_handler "-i|--instance_dir: Given path is not a zowe instance directory"
+                fi
+            else
+                error_handler "-i|--instance_dir: Given path is not a zowe instance directory or does not exist"
+            fi
+            shift
+        ;;
+        -n|--native)
+            shift
+            IS_NATIVE=$1
+            shift
+        ;;
+        -d|--target_dir) # Represents the path to the desired target directory to place the extensions (optional)
+            shift
+            TARGET_DIR=$(get_full_path "$1")
+            shift
+        ;;
+        -l|--logs-dir) # Represents the path to the installation logs
+            shift
+            LOG_DIRECTORY=$1
+            shift
+        ;;
+        *)
+            error_handler "$1 is an invalid flag\ntry: zowe-configure-component.sh -o {COMPONENT_NAME} -i {ZOWE_INSTANCE_DIR}"
+            shift
+    esac
+done
+
+if [ -z $TARGET_DIR ]; then
+    TARGET_DIR=$DEFAULT_TARGET_DIR
+fi
```

Comment:
We need to validate TARGET_DIR matches ZWE_EXTENSION_DIR defined in instance.env or not. If they don't match, then we need to reject the configure step.